Update references to hEngine across our sites (#1052)
hashintel/hash: sites/hashai/resources/docs/simulation/creating-simulations/behaviors/inheritance.md

We're now planning to introduce support for **Agent Types** in a forthcoming update, which go further. In the future, [HASH Engine](/platform/engine) (hEngine) will attempt to match the _type_ field of an agent to an entity type in HASH, where it contains the URI of an _entity type_ schema.

Entity types created in HASH describe 'things'. In a simulation context these could for example be:

1.  agents \(Agent Types\); or
1.  actions taken by agents \(Behaviors\).

Entity types consist of _properties_, each of which has their _own_ property types, and in some cases these may have _default values_. Entity types in HASH may contain a `hash_simulation_behavior` property containing one or more [behaviors](/docs/simulation/creating-simulations/behaviors).

When an agent in a HASH simulation has its _type_ declared going forward, if hEngine is able to match that type to a valid entity type URI, any behaviors specified in that schema's `hash_simulation_behavior` array will be attached to the agent.

There is no limit to the number of types that may be attached to an agent, and types may continue to be used descriptively without being linked to entity type URIs.

<Hint style="info">

> As part of these changes, we'll be allowing users to attach arrays of HASH simulation behavior files to entity types under the protected `hash_simulation_behavior` property type field.

Reviewer: I wonder if we want to use the term "list" instead of "array"? it sounds less technical.

Author: Implemented here: https://github.com/hashintel/hash/pull/1052/commits/775d9bd2f0bfe5ee0fc372edcef90f0daabfc31f

Reviewer: I think that is better, thanks 👍

> As part of these changes, we'll be allowing users to attach arrays of HASH simulation behavior files to entity types under the protected `hash_simulation_behavior` property type field.

Author: ```suggestion
As part of these changes, you'll be able to attach any number of HASH simulation behavior files to an entity type under the protected `hash_simulation_behavior` property.
```

Author: As suggested by Tim here: https://github.com/hashintel/hash/pull/1052#discussion_r967716943
